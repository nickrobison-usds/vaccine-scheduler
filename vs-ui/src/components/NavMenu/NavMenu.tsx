import { Link, NavLink } from "react-router-dom";
import { Header, PrimaryNav, Title } from "@trussworks/react-uswds";
import React from "react";

export const NavMenu: React.FC<{}> = () => {
  const navItems = [
    <NavLink key="home" to="/">
      Home
    </NavLink>,
    <NavLink key="validate" to="/validate">
      Validate
    </NavLink>,
    <NavLink key="finder" to="/finder">
      Finder
    </NavLink>,
  ];

  return (
    <Header basic>
      <div className="usa-nav-container">
        <div className="usa-navbar">
          <Title>
            <Link to="/">VaccineScheduler</Link>
          </Title>
        </div>
        <PrimaryNav aria-label="Primary navigation" items={navItems} />
      </div>
    </Header>
  );
};
