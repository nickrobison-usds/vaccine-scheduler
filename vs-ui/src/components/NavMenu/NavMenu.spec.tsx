import {NavMenu} from "./NavMenu";
import {render, screen} from "@testing-library/react";
import React from "react";
import {MemoryRouter} from "react-router-dom";

const TestContainer: React.FC = ({children}) => (
    <MemoryRouter>
        <>{children}</>
    </MemoryRouter>
);

describe('NavMenu', () => {
    it('renders correctly', () => {
        const {container} = render(<TestContainer><NavMenu/></TestContainer>);
        expect(screen.getByText('Finder')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    })
});
