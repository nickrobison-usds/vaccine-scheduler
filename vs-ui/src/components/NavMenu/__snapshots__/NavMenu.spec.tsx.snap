// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu renders correctly 1`] = `
<div>
  <header
    class="usa-header usa-header--basic"
    data-testid="header"
  >
    <div
      class="usa-nav-container"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
        >
          <em
            class="usa-logo__text"
          >
            <a
              href="/"
            >
              VaccineScheduler
            </a>
          </em>
        </div>
      </div>
      <nav
        aria-label="Primary navigation"
        class="usa-nav"
      >
        <button
          class="usa-nav__close"
          data-testid="navCloseButton"
          type="button"
        >
          <img
            alt="close"
            src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0IiB2aWV3Qm94PSIwIDAgNjQgNjQiPjx0aXRsZT5jbG9zZTwvdGl0bGU+PHBhdGggZmlsbD0iIzU2NUM2NSIgZmlsbC1ydWxlPSJldmVub2RkIiBkPSJNNTcuMDQyIDEuMTVsNS44MDkgNS44MDhhNCA0IDAgMCAxIDAgNS42NTdMNDMuNDY1IDMybDE5LjM4NiAxOS4zODVhNCA0IDAgMCAxIDAgNS42NTdsLTUuODA5IDUuODA5YTQgNCAwIDAgMS01LjY1NyAwTDMyIDQzLjQ2NSAxMi42MTUgNjIuODUxYTQgNCAwIDAgMS01LjY1NyAwbC01LjgwOS01LjgwOWE0IDQgMCAwIDEgMC01LjY1N0wyMC41MzUgMzIgMS4xNDkgMTIuNjE1YTQgNCAwIDAgMSAwLTUuNjU3bDUuODA5LTUuODA5YTQgNCAwIDAgMSA1LjY1NyAwTDMyIDIwLjUzNSA1MS4zODUgMS4xNDlhNCA0IDAgMCAxIDUuNjU3IDB6Ii8+PC9zdmc+"
          />
        </button>
        <ul
          class="usa-nav__primary usa-accordion"
        >
          <li
            class="usa-nav__primary-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="usa-nav__primary-item"
          >
            <a
              href="/finder"
            >
              Finder
            </a>
          </li>
          <li
            class="usa-nav__primary-item"
          >
            <a
              href="/sources"
            >
              Sources
            </a>
          </li>
          <li
            class="usa-nav__primary-item"
          >
            <a
              href="/validate"
            >
              Validate
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
